vm/actor/src/builtin/market/mod.rs

```diff
@@ -4,7 +4,687 @@
+        rt: &mut RT,
+        params: VerifyDealsOnSectorProveCommitParams,
+    ) -> Result<DealWeight, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_type(std::iter::once(&*MINER_ACTOR_CODE_ID))?;
+        let miner_addr = rt.message().from().clone();
+        let mut total_deal_space_time = BigInt::zero();
+        let mut deal_weight = BigInt::zero();
+
+        rt.transaction::<State, Result<(), ActorError>, _>(|st, rt| {
+            // if there are no dealIDs, it is a CommittedCapacity sector
+            // and the totalDealSpaceTime should be zero
+            let mut states = Amt::load(&st.states, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?;
+            let proposals = Amt::load(&st.proposals, rt.store())
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?;
+
+            for id in &params.deal_ids {
+                let mut deal: DealState = states
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(ExitCode::ErrIllegalState, "Get deal error".to_owned())
+                    })?;
+                let proposal: DealProposal = proposals
+                    .get(*id)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalArgument, e.into()))?
+                    .ok_or_else(|| {
+                        ActorError::new(ExitCode::ErrIllegalState, "Get deal error".to_owned())
+                    })?;
+
+                validate_deal_can_activate(
+                    rt.curr_epoch(),
+                    &miner_addr,
+                    params.sector_expiry,
+                    &proposal,
+                )?;
+
+                deal.sector_start_epoch = OptionalEpoch(Some(rt.curr_epoch()));
+                states
+                    .set(*id, deal)
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+                // compute deal weight
+                let deal_space_time = proposal.duration() * proposal.piece_size.0;
+                total_deal_space_time += deal_space_time;
+            }
+            st.states = states
+                .flush()
+                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+            let epoch_value = params
+                .sector_expiry
+                .checked_sub(rt.curr_epoch())
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalArgument,
+                        format!(
+                            "invalid sector expiry epoch {:?} ahead of current {:?}",
+                            params.sector_expiry,
+                            rt.curr_epoch()
+                        ),
+                    )
+                })?;
```

Comment:
oops I missed this, you didn't change this. To be precise, it needs to be:
```
let epoch_value = params
                .sector_expiry
                .checked_sub(rt.curr_epoch())
                .unwrap_or(1u8.into());
```

Reply:
This is changed? 

```
let epoch_value = params
                .sector_expiry
                .checked_sub(rt.curr_epoch())
                .ok_or(1u8) 
                .map_err(|_| {
                    ActorError::new(
                        ExitCode::ErrIllegalArgument,
                        format!(
                            "invalid sector expiry epoch {:?} ahead of current {:?}",
                            params.sector_expiry,
                            rt.curr_epoch()
                        ),
                    )
                })?;
```

Reply:
nah, what you're doing there is converting the `Option<ChainEpoch>` to `Result<ChainEpoch, u8>` then unwrapping that and ignoring the u8 value to throw the error anyway

Reply:
sorry if my previous comment was misleading, it was just meant as pseudocode